resources/services/network/interfaces.go

```diff
@@ -0,0 +1,407 @@
+			},
+			{
+				Name:        "id",
+				Description: "Resource ID.",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("ID"),
+			},
+			{
+				Name:        "location",
+				Description: "Resource location.",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "name",
+				Description: "Resource name.",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "dns_settings_applied_dns_servers",
+				Description: "The servers that are part of the same availability set.",
+				Type:        schema.TypeStringArray,
+				Resolver:    schema.PathResolver("InterfacePropertiesFormat.DNSSettings.AppliedDNSServers"),
+			},
+			{
+				Name:        "dns_settings_dns_servers",
+				Description: "List of DNS servers IP addresses.",
+				Type:        schema.TypeStringArray,
+				Resolver:    schema.PathResolver("InterfacePropertiesFormat.DNSSettings.DNSServers"),
+			},
+			{
+				Name:        "dns_settings_internal_dns_name_label",
+				Description: "The internal dns name label.",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("InterfacePropertiesFormat.DNSSettings.InternalDNSNameLabel"),
+			},
+			{
+				Name:        "dns_settings_internal_domain_name_suffix",
+				Description: "The internal domain name suffix.",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("InterfacePropertiesFormat.DNSSettings.InternalDomainNameSuffix"),
+			},
+			{
+				Name:        "dns_settings_internal_fqdn",
+				Description: "Fully qualified DNS name supporting internal communications between VMs in the same virtual network.",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("InterfacePropertiesFormat.DNSSettings.InternalFqdn"),
+			},
+
+			{
+				Name:        "dscp_configuration_id",
+				Description: "A reference to the dscp configuration to which the network interface is linked.",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("InterfacePropertiesFormat.DscpConfiguration.ID"),
+			},
+			{
+				Name:        "enable_accelerated_networking",
+				Description: "If the network interface is accelerated networking enabled.",
+				Type:        schema.TypeBool,
+				Resolver:    schema.PathResolver("InterfacePropertiesFormat.EnableAcceleratedNetworking"),
+			},
+			{
+				Name:        "enable_ip_forwarding",
+				Description: "Indicates whether IP forwarding is enabled on this network interface.",
+				Type:        schema.TypeBool,
+				Resolver:    schema.PathResolver("InterfacePropertiesFormat.EnableIPForwarding"),
+			},
+			{
+				Name:        "hosted_workloads",
+				Description: "List of references to linked BareMetal resources.",
+				Type:        schema.TypeStringArray,
+				Resolver:    schema.PathResolver("InterfacePropertiesFormat.HostedWorkloads"),
+			},
+			{
+				Name:        "mac_address",
+				Description: "The MAC address of the network interface.",
+				Type:        schema.TypeString,
```

Comment:
nit: would suggest to maybe use schema.TypeMacAddr? you can use the schema.MacAddressResolver

Reply:
might just keep them like this for now and see how things go with iterations. Run into some problems with database types after changing.